core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -304,25 +301,26 @@ class AclAuthorizer extends Authorizer with Logging {
   }
 
   private def matchingAcls(resourceType: ResourceType, resourceName: String): Set[Acl] = {
-    inReadLock(lock) {
-      val wildcard = aclCache.get(Resource(resourceType, Acl.WildCardResource, PatternType.LITERAL))
-        .map(_.acls)
-        .getOrElse(Set.empty[Acl])
-
-      val literal = aclCache.get(Resource(resourceType, resourceName, PatternType.LITERAL))
-        .map(_.acls)
-        .getOrElse(Set.empty[Acl])
-
-      val prefixed = aclCache
-        .from(Resource(resourceType, resourceName, PatternType.PREFIXED))
-        .to(Resource(resourceType, resourceName.take(1), PatternType.PREFIXED))
-        .filterKeys(resource => resourceName.startsWith(resource.name))
-        .values
-        .flatMap { _.acls }
-        .toSet
-
-      prefixed ++ wildcard ++ literal
-    }
+    // snapshot immutable aclCache to avoid the need for a readLock that
```

Comment:
Nit: the way this is written, it sounds like we are taking a snapshot. But all we are doing is saving a reference to an immutable structure in a local val so that we get a consistent view. Right?

Reply:
I've updated the comment.